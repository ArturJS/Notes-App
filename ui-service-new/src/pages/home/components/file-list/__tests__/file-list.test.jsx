jest.mock('../components/files-list-item', () => props => (
    <div className="files-list-item">
        {JSON.stringify(
            props,
            (prop, value) => {
                if (typeof value === 'function') {
                    return `function ${prop}`;
                }

                return value;
            },
            '  '
        )}
    </div>
));

/* eslint-disable import/first */
import React from 'react';
import renderer from 'react-test-renderer';
import FilesList from '../file-list';
/* eslint-enable import/first */

describe('<FilesList />', () => {
    it('renders nothing if no files', () => {
        const tree = renderer.create(<FilesList files={[]} />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly list of files', () => {
        const files = [
            {
                id: 1,
                name: 'File 1'
            },
            {
                id: 2,
                name: 'File 2',
                remove: () => {}
            }
        ];
        const tree = renderer.create(<FilesList files={files} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
