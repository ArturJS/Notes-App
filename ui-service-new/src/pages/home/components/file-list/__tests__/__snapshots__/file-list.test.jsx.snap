// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilesList /> renders correctly list of files 1`] = `
<ul
  className="files-list list-unstyled"
>
  <div
    className="files-list-item"
  >
    {
  "file": {
    "id": 1,
    "name": "File 1"
  }
}
  </div>
  <div
    className="files-list-item"
  >
    {
  "file": {
    "id": 2,
    "name": "File 2",
    "remove": "function remove"
  },
  "onRemove": "function onRemove"
}
  </div>
</ul>
`;

exports[`<FilesList /> renders nothing if no files 1`] = `null`;
